sdk/storage/azure-storage-blob/tests/test_blob_tags_async.py

```diff
@@ -376,4 +376,36 @@ async def test_filter_blobs(self, storage_account_name, storage_account_key):
         self.assertEqual(len(items_on_page2[0]['tags']), 2)
         self.assertEqual(items_on_page2[0]['tags']['tag1'], 'firsttag')
         self.assertEqual(items_on_page2[0]['tags']['tag2'], 'secondtag')
+
+    @BlobPreparer()
+    async def test_filter_blobs_versions(self, versioned_storage_account_name, versioned_storage_account_key):
+        await self._setup(versioned_storage_account_name, versioned_storage_account_key)
+
+        blob_name = self._get_blob_reference()
+        blob_client = self.bsc.get_blob_client(self.container_name, blob_name)
+        tags1 = {"tag1": "firsttag"}
+        include_list = ['versions']
+        where = "\"tag1\"='firsttag'"
+
+        # Act
+        await blob_client.create_append_blob(tags=tags1)
+        await blob_client.create_append_blob(tags=tags1)
+
+        if self.is_live:
+            sleep(10)
+
+        blob_list = self.bsc.find_blobs_by_tags(filter_expression=where, results_per_page=2,
```

Comment:
Same comment about refactoring to not use `by_page`. It will be slightly different for async, you'll have to use a separate list variable an `async for` to append all items to the list then do the asserts. There should be plenty of examples around, you can look at `list_blobs` examples.